includes/CLI/Plugin_List_Check_Categories_Command.php

```diff
@@ -0,0 +1,179 @@
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\CLI;
+
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Plugin_Context;
+use WP_CLI;
+
+/**
+ * Plugin list check categories command.
+ */
+final class Plugin_List_Check_Categories_Command {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since n.e.x.t
+	 * @var Plugin_Context
+	 */
+	protected $plugin_context;
+
+	/**
+	 * Output format type.
+	 *
+	 * @since n.e.x.t
+	 * @var string[]
+	 */
+	protected $output_formats = array(
+		'table',
+		'csv',
+		'json',
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Plugin_Context $plugin_context Plugin context.
+	 */
+	public function __construct( Plugin_Context $plugin_context ) {
+		$this->plugin_context = $plugin_context;
+	}
+
+	/**
+	 * Runs plugin list-check-categories.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--fields=<fields>]
+	 * : Limit displayed results to a subset of fields provided.
+	 *
+	 * [--format=<format>]
+	 * : Format to display the results. Options are table, csv, and json. The default will be a table.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   wp plugin list-check-categories
+	 *   wp plugin list-check-categories --format=json
+	 *
+	 * @subcommand list-check-categories
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args       List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 */
+	public function list_check_categories( $args, $assoc_args ) {
+		// Get options based on the CLI arguments.
+		$options = $this->get_options( $assoc_args );
+
+		// Get categories details.
+		$categories = $this->get_categories();
+
+		// Get formatter.
+		$formatter = $this->get_formatter( $options );
+
+		// Display results.
+		$formatter->display_items( $categories );
+	}
+
+	/**
+	 * Returns check categories details.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of the check categories.
+	 */
+	private function get_categories() {
+		$categories = array();
+
+		$check_categories = new Check_Categories();
+		$categories_slugs = $check_categories->get_categories();
+
+		if ( is_array( $categories_slugs ) && count( $categories_slugs ) > 0 ) {
+			$categories = array_map(
+				function ( $slug ) {
+					return array(
+						'slug' => $slug,
+						'name' => ucfirst( str_replace( '_', ' ', $slug ) ),
+					);
+				},
+				$categories_slugs
+			);
+
+		}
+
+		return $categories;
+	}
+
+	/**
+	 * Validates the associative arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if plugin not found.
+	 */
+	private function get_options( $assoc_args ) {
```

Comment:
This method could still be moved over to `Plugin_Check_Command` by renaming it to be more specific, e.g. `get_list_check_categories_options()`.

We should probably rename `Plugin_Check_Command::get_options()` to `Plugin_Check_Command::get_check_options()` accordingly so that all 3 methods are scoped to the command they are for.